lib/mix/tasks/phx.gen.html.ex

```diff
@@ -176,13 +176,13 @@ defmodule Mix.Tasks.Phx.Gen.Html do
       {key, :text} ->
         {label(key), ~s(<%= textarea f, #{inspect(key)} %>), error(key)}
       {key, :date} ->
-        {label(key), ~s(<%= date_select f, #{inspect(key)} %>), error(key)}
+        {label(key), ~s(<%= date_input f, #{inspect(key)} %>), error(key)}
       {key, :time} ->
-        {label(key), ~s(<%= time_select f, #{inspect(key)} %>), error(key)}
+        {label(key), ~s(<%= time_input f, #{inspect(key)} %>), error(key)}
       {key, :utc_datetime} ->
-        {label(key), ~s(<%= datetime_select f, #{inspect(key)} %>), error(key)}
+        {label(key), ~s(<%= datetime_local_input f, #{inspect(key)} %>), error(key)}
```

Comment:
I gave it a try locally and it seems that `datetime_local_input` is not supported in Firefox nor Safari. Nor `datetime_input`. We may have to use another aproach to solve the date/time/datetime issues, possibly keeping them as `datetime_select` and use maps in tests as you proposed as alternative. Sorry for misleading. :(

Reply:
Damm browsers supports 😢 

ok, I will make the change for using maps on the tests instead 👍